<launch>
  <arg name="node_name" default="morphology" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />
  <arg name="queue_size" default="3" doc="Specigy queue_size of input image subscribers" />

  <arg name="morph_operator" default="0" doc="Morphology Operation Methods  0: Opening, 1: Closing, 2: Gradient, 3: Top Hat, 4: Black Hat" />
  <arg name="morph_element" default="0" doc="Type of the kernel 0: Rect, 1: Cross, 2: Ellipse" />
  <arg name="morph_size" default="0" doc="Size of the kernel" />

  <!-- morphology.cpp  -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="morphology" >
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="queue_size" value="$(arg queue_size)" />
    <param name="morph_operator" value="$(arg morph_operator)" />
    <param name="morph_element" value="$(arg morph_element)" />
    <param name="morph_size" value="$(arg morph_size)" />
  </node>
</launch>
